crates/node/src/service/rpc/mod.rs

```diff
@@ -0,0 +1,110 @@
+use std::sync::Arc;
+
+use jsonrpsee::server::ServerHandle;
+use tokio::task::JoinSet;
+
+use mc_db::DatabaseService;
+use mc_rpc::{providers::AddTransactionProvider, rpc_api_admin, rpc_api_user, Starknet};
+use mp_utils::service::Service;
+
+use metrics::RpcMetrics;
+use server::{start_server, ServerConfig};
+
+use crate::cli::{RpcEndpoints, RpcParams};
+
+use self::server::rpc_api_build;
+
+mod metrics;
+mod middleware;
+mod server;
+
+pub struct RpcService {
+    server_config_user: Option<ServerConfig>,
+    server_config_admin: Option<ServerConfig>,
+    server_handle_user: Option<ServerHandle>,
+    server_handle_admin: Option<ServerHandle>,
+}
+impl RpcService {
+    pub fn new(
+        config: &RpcParams,
+        db: &DatabaseService,
+        add_txs_method_provider: Arc<dyn AddTransactionProvider>,
+    ) -> anyhow::Result<Self> {
+        if config.rpc_disabled {
+            return Ok(Self {
+                server_config_user: None,
+                server_config_admin: None,
+                server_handle_user: None,
+                server_handle_admin: None,
+            });
+        }
+
+        let (rpcs, node_operator) = match config.rpc_endpoints {
+            RpcEndpoints::Safe => (true, false),
+            RpcEndpoints::Auto | RpcEndpoints::Unsafe => (true, true),
+        };
+
+        let (read, write, trace, admin, ws) = (rpcs, rpcs, rpcs, node_operator, rpcs);
```

Comment:
can you remove this atrocity
(it's my fault originally so i'm allowed to say that ahah)